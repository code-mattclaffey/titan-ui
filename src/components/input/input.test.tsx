import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Input } from '.';

afterEach(() => {
    cleanup();
});

describe('<Input />', () => {
    it('should match snapshot', () => {
        const { container } = render(
            <Input
                type="text"
                name="hello-test"
                id="hello-test"
                labelText="Hello World"
            />,
        );

        expect(container).toMatchSnapshot();
    });

    it('should match snapshot when it has a value', () => {
        const { container } = render(
            <Input
                type="text"
                name="hello-test"
                id="hello-test"
                defaultValue="hello world"
                labelText="Hello World"
            />,
        );

        expect(container).toMatchSnapshot();
    });
});
