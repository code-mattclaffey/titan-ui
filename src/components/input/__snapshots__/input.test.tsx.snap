// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
<div>
  <div
    class="e-input__container"
  >
    <label
      class="e-label e-label--required"
      data-testid="e-label"
      for="hello-test"
    >
      <span
        class="e-label__text"
      >
        Hello World
      </span>
      <input
        class="e-input e-input--empty"
        id="hello-test"
        name="hello-test"
        required=""
        type="text"
        value=""
      />
    </label>
  </div>
</div>
`;

exports[`<Input /> should match snapshot when it has a value 1`] = `
<div>
  <div
    class="e-input__container"
  >
    <label
      class="e-label e-label--required"
      data-testid="e-label"
      for="hello-test"
    >
      <span
        class="e-label__text"
      >
        Hello World
      </span>
      <input
        class="e-input"
        id="hello-test"
        name="hello-test"
        required=""
        type="text"
        value="hello world"
      />
    </label>
  </div>
</div>
`;
